lobster/tools/codebeamer/codebeamer.py

```diff
@@ -373,108 +378,102 @@ def ensure_array_of_strings(instance):
-
     ap.add_argument("--config",
-                    help=("name of codebeamer "
-                          "config file, supported references: '%s'" %
+                    help=("Path to YAML file with arguments, "
+                          "by default (codebeamer-config.yaml) "
+                          "supported references: '%s'" %
                           ', '.join(SupportedConfigKeys.as_set())),
-                    default=None)
-
-    ap.add_argument("--ignore-ssl-errors",
-                    action="store_true",
-                    default=False,
-                    help="ignore ssl errors and accept any certificate")
-
-    ap.add_argument("--query-size",
-                    type=int,
-                    default=100,
-                    help=("Fetch this many cb items at once (by default 100),"
-                          " reduce if you get too many timeouts."))
-
-    ap.add_argument("--timeout",
-                    type=int,
-                    default=30,
-                    help="Timeout in s (by default 30) for each REST call.")
-
-    ap.add_argument("--schema",
-                    default='requirement',
-                    help="Specify the output schema"
-                         "(Requirement, Implementation, Activity).")
-
-    ap.add_argument("--cb-root", default=os.environ.get("CB_ROOT", None))
-    ap.add_argument("--cb-user", default=os.environ.get("CB_USERNAME", None))
-    ap.add_argument("--cb-pass", default=os.environ.get("CB_PASSWORD", None))
-    ap.add_argument("--cb-token", default=None)
-    ap.add_argument("--out", default=None)
+                    default=os.path.join(os.getcwd(), "codebeamer-config.yaml"))
+
+    ap.add_argument("--out",
+                    help=("Name of output file"),
+                    default="codebeamer.lobster")
+
     options = ap.parse_args()
 
     mh = Message_Handler()
 
-    cb_config = {
-        'schema'     : options.schema,
-        "root"       : options.cb_root,
-        "base"       : "%s/cb/api/v3" % options.cb_root,
-        "user"       : options.cb_user,
-        "pass"       : options.cb_pass,
-        "token"      : options.cb_token,
-        "verify_ssl" : not options.ignore_ssl_errors,
-        "page_size"  : options.query_size,
-        "timeout"    : options.timeout,
-    }
+    if not os.path.isfile(options.config):
+        print((f"lobster-codebeamer: Config file '{options.config}' not found."))
+        return 1
 
-    if options.config:
-        if os.path.isfile(options.config):
-            cb_config.update(parse_cb_config(options.config))
-        else:
-            ap.error("cannot open config file '%s'" % options.config)
+    cb_config = parse_yaml_config(options.config)
+
+    if cb_config["out"] is None:
+        cb_config["out"] = options.out
 
     if cb_config["root"] is None:
-        ap.error("please set CB_ROOT or use --cb-root")
+        sys.exit("lobster-codebeamer: Please set 'root' in the config file")
```

Comment:
I observed that Argument parser error was replaced by sys.exit. Any specific reason for that?

Reply:
Because we are switching from command-line arguments to a config YAML file, it is essentially an argument parser error.